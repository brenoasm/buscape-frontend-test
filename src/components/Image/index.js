import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  onBrokenImageUrl: PropTypes.func,
  width: PropTypes.number,
  height: PropTypes.number
};

const defaultProps = {
  src: "",
  alt: "Imagem",
  width: 40,
  height: 40
};

const StyledImage = styled.div`
  height: ${props => props.height}px;
  width: ${props => props.width}px;

  > img {
    height: ${props => props.height}px;
    width: ${props => props.width}px;
  }
`;

// defaultProp do onBrokenImageUrl está sendo setado aqui, pois dividir na constante separada está causando um erro
const Image = ({ src, alt, width, height, onBrokenImageUrl = null }) => (
  <StyledImage width={width} height={height}>
    <img src={src} alt={alt} onError={onBrokenImageUrl} />
  </StyledImage>
);

StyledImage.propTypes = propTypes;
StyledImage.defaultProps = defaultProps;

export default Image;
