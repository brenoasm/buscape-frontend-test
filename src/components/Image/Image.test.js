import React from "react";
import { shallow } from "enzyme";

import Image from "./";

describe("Image", () => {
  it("matches the snapshot", () => {
    const imageProps = {
      src:
        "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
      alt: "Teste",
      width: 80,
      height: 80,
      onBrokenImageUrl: jest.fn()
    };

    const wrapper = shallow(<Image {...imageProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
