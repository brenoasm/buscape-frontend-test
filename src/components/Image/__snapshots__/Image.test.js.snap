// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="Teste"
    height={80}
    onBrokenImageUrl={[MockFunction]}
    src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
    width={80}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "alt": "Imagem",
      "children": <img
        alt="Teste"
        onError={[MockFunction]}
        src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
      />,
      "height": 80,
      "src": "",
      "width": 80,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Teste",
        "onError": [MockFunction],
        "src": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  height: ",
          [Function],
          "px;
  width: ",
          [Function],
          "px;

  > img {
    height: ",
          [Function],
          "px;
    width: ",
          [Function],
          "px;
  }
",
        ],
      },
      "defaultProps": Object {
        "alt": "Imagem",
        "height": 40,
        "src": "",
        "width": 40,
      },
      "displayName": "styled.div",
      "propTypes": Object {
        "alt": [Function],
        "height": [Function],
        "onBrokenImageUrl": [Function],
        "src": [Function],
        "width": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "alt": "Imagem",
        "children": <img
          alt="Teste"
          onError={[MockFunction]}
          src="http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg"
        />,
        "height": 80,
        "src": "",
        "width": 80,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Teste",
          "onError": [MockFunction],
          "src": "http://thumbs.buscape.com.br/celular-e-smartphone/smartphone-apple-iphone-7-128gb_600x600-PU98460_1.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  height: ",
            [Function],
            "px;
  width: ",
            [Function],
            "px;

  > img {
    height: ",
            [Function],
            "px;
    width: ",
            [Function],
            "px;
  }
",
          ],
        },
        "defaultProps": Object {
          "alt": "Imagem",
          "height": 40,
          "src": "",
          "width": 40,
        },
        "displayName": "styled.div",
        "propTypes": Object {
          "alt": [Function],
          "height": [Function],
          "onBrokenImageUrl": [Function],
          "src": [Function],
          "width": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
